src/services/legal-services.ts

```diff
@@ -133,6 +133,30 @@ export default class LegalServices {
       })
   }
 
+  /**
+   * Fetches name request data with phone and email validation.
+   * @param nrNumber the name request number (eg, NR 1234567)
+   * @param phone the name request phone (eg, 12321232)
+   * @param email the name request email (eg, ab@gmail.com)
+   * @returns a promise to return the NR data, or null if not found
+   */
+  static async fetchValidContactNr (nrNumber: string, phone?: string, email?: string): Promise<NameRequestIF> {
```

Comment:
Instead of simply making phone and email optional (and thus possibly undefined), I recommend assigning them a default value instead, eg
```
static async fetchValidContactNr (nrNumber: string, phone = '', email = '') Promise<NameRequestIF> {
```